### PR TITLE
Update package name, tsconfig, license, homepage link, exported files

### LINKED ISSUE #31: Publish the Safe Token NPM Package
### Expected Outcome

- One or more PRs to fix small details in the `package.json`:
  - License is incorrect and not a valid SPDX identifier
  - Package name should be org scoped
  - exported files
  - etc. (Go over <https://docs.npmjs.com/cli/v9/configuring-npm/package-json> to see if there is anything else)
- A package deployed on NPM.

### PR DESCRIPTION
**Pull request by akshay-ap**
Fixes #31 

Changes in PR:
- Package name includes org name
- Update homepage link
- Use licence id `LGPL-3.0-only`
- Remove `test` and `src` from files
- Update tsconfig so that `/src` directory is included

### CODE REVIEW

**Review by mmv08 — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "node_modules"
   ],
   "include": [
-    "./src/index.ts",
-    "./types"
+    "./types",
+    "./src"
```
Comment (remedcu) [inline tsconfig.json:28]:
```suggestion
    "./src",
    "./types"
```

uber-nit: Alphabetical order

File: package.json [hunk 2]
```diff
@@ -1,14 +1,12 @@
 {
-  "name": "safe-token",
+  "name": "@safe-global/safe-token",
   "version": "1.0.0",
   "description": "Repository for Safe token related contracts",
   "homepage": "https://www.safe.global",
```
Comment (remedcu) [inline package.json:5]:
```suggestion
  "homepage": "https://safe.global",
```

We don't use `www` anymore right?

File: package.json [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-  "license": "GPL-3.0",
+  "license": "LGPL-3.0",
```
Comment (nlordell) [inline package.json:6]:
```suggestion
  "license": "LGPL-3.0-only",
```

The [`LGPL-3.0` identifier is deprecated](https://spdx.org/licenses/LGPL-3.0.html).

File: package.json [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
   "files": [
     "contracts",
     "dist",
```
Comment (nlordell) [inline package.json:9]: I don't think we need to include the `dist` directory. There are no TS libraries that we actually export with this project IMO.

File: package.json [hunk 5]
```diff
@@ -1,14 +1,12 @@
...
-    "src",
-    "test",
     "build"
```
Comment (nlordell) [inline package.json:10]: This means that we include the build cache with the package, which I don't think we should:

```suggestion
    "build/artifacts"
```

**Pull request merged**